packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -149,12 +152,13 @@ function getDelta(
   // is portaled somewhere other than the body and has an ancestor with
   // position: relative/absolute, it will be different.
   containerDimensions: Dimensions,
-  padding: number
+  padding: number,
+  containerOffsetWithBoundary: Offset
 ) {
   let containerScroll = containerDimensions.scroll[axis];
   let boundaryHeight = boundaryDimensions[AXIS_SIZE[axis]];
-  let startEdgeOffset = offset - padding - containerScroll;
-  let endEdgeOffset = offset + padding - containerScroll + size;
+  let startEdgeOffset = offset - padding - containerScroll + containerOffsetWithBoundary[axis];
+  let endEdgeOffset = offset + padding - containerScroll + size + containerOffsetWithBoundary[axis];
```

Comment:
In the case where the menu is portaled to a container that isn't the body, we need to adjust the offset calculated to fit the same coordinate system that the boundary uses hence the addition of the `containerOffsetWIthBounday[axis]` here. This fixes the `getDelta` for submenus since they are positioned with respect to their parent menu, not the page body.